public/components/utils/EditableDropdown.js

```diff
@@ -0,0 +1,73 @@
+import React, { PropTypes } from 'react';
+
+class EditableDropdown extends React.Component {
+
+  static propTypes = {
+    name: PropTypes.string.isRequired,
+    values: PropTypes.array.isRequired,
+    selectedValue: PropTypes.string.isRequired,
+    onChange: PropTypes.func.isRequired
+  };
+
+  state = {
+    editable: false
+  }
+
+  enableEditing = () => {
+    this.setState({
+      editable: true
+    });
+  }
+
+  disableEditing = () => {
+    this.setState({
+      editable: false
+    });
+  }
+
+  prepareOptions = () => {
+    const values = this.props.values;
+    let options = [];
+
+    values.map((prop, index) =>
+      options.push(<option value={prop.value} key={index}>{prop.name}</option>)
+    );
+
```

Comment:
So this could be written as 

```
return values.map((prop, index) =>
      return <option value={prop.value} key={prop.value}>{prop.name}</option>
 );
```